centreon-certified/omi/omi_events-apiv2.lua

```diff
@@ -0,0 +1,363 @@
+function EventQueue:format_event_service()
+  local service_severity = self.sc_broker:get_severity(self.sc_event.event.host_id, self.sc_event.event.service_id)
+
+  if service_severity == false then
+    service_severity = 0
+  end
+
+  self.sc_event.event.formated_event = {
+    title = self.sc_event.event.cache.service.description,
+    description =  string.match(self.sc_event.event.output, "^(.*)\n"),
+    severity = self.sc_event.event.state,
+    time_created = self.sc_event.event.last_update,
+    node = self.sc_event.event.cache.host.name,
+    related_ci = self.sc_event.event.cache.host.name,
+    source_ci = self.sc_common:ifnil_or_empty(self.source_ci, 'Centreon'),
+    source_event_id = self.sc_common:ifnil_or_empty(self.sc_event.event.service_id, 0)
+  }
+end
+
+--------------------------------------------------------------------------------
+-- EventQueue:add method
+-- @param e An event
+--------------------------------------------------------------------------------
+
+function EventQueue:add()
+  -- store event in self.events lists
+  local category = self.sc_event.event.category
+  local element = self.sc_event.event.element
+
+  self.sc_logger:debug("[EventQueue:add]: add event in queue category: " .. tostring(self.sc_params.params.reverse_category_mapping[category])
+    .. " element: " .. tostring(self.sc_params.params.reverse_element_mapping[category][element]))
+
+  self.sc_logger:debug("[EventQueue:add]: queue size before adding event: " .. tostring(#self.sc_flush.queues[category][element].events))
+  self.sc_flush.queues[category][element].events[#self.sc_flush.queues[category][element].events + 1] = self.sc_event.event.formated_event
+
+  self.sc_logger:info("[EventQueue:add]: queue size is now: " .. tostring(#self.sc_flush.queues[category][element].events)
+    .. "max is: " .. tostring(self.sc_params.params.max_buffer_size))
+end
+
+--------------------------------------------------------------------------------
+-- EventQueue:build_payload, concatenate data so it is ready to be sent
+-- @param payload {string} xml encoded string
+-- @param event {table} the event that is going to be added to the payload
+-- @return payload {string} xml encoded string
+--------------------------------------------------------------------------------
+function EventQueue:build_payload(payload, event)
+  if not payload then
+    payload = "<event_data>\t"
+    for index, xml_str in pairs(event) do
+      payload = payload .. "<" .. tostring(index) .. ">" .. tostring(self.sc_common:xml_escape(xml_str)) .. "</" .. tostring(index) .. ">\t"
+    end
+    payload = payload .. "</event_data>"
+
+  else
+    payload = payload .. "\n<event_data>\t"
+    for index, xml_str in pairs(event) do
+      payload = payload .. "<" .. tostring(index) .. ">" .. tostring(self.sc_common:xml_escape(xml_str)) .. "</" .. tostring(index) .. ">\t"
+    end
+    payload = payload .. "</event_data>"
+  end
+
+  return payload
+end
+
```

Comment:
@tanguyvda said : "you may have to take a closer look at the old stream connector. It uses ltn12 and you don't"

Reply:
If I understood the use of ltn12 in the old code, it would be to store the event in a table and send it in the API request. Basically, it's the body of the request. Then we use another system in our send_data function.

Here are the references I found:
- ltn12.source.string :
https://github.com/lunarmodules/luasocket/blob/5b18e475f38fcf28429b1cc4b17baee3b9793a62/doc/ltn12.html#L399

- ltn12.sink.table :
https://github.com/lunarmodules/luasocket/blob/5b18e475f38fcf28429b1cc4b17baee3b9793a62/doc/ltn12.html#L254